### PR TITLE
fix: preserves tools inner in local drawer in AppLayout toolbar variant

### PR DESCRIPTION
**Pull request by dpitcock**
### Description

- preserves tools inner content in local drawer in AppLayout toolbar variant
- removes skipping condition for unit test for runtime drawers
- adds integ test to the stateful page

- not relevant to this fix but in the pull request there is also the addition of a testing util `testOnlyIf()` to be used when debugging integ tests locally.

Human testing Go to https://d21d5uik3ws71m.cloudfront.net/components/ca007ca1f62430c066401d68d11d4ded65cbe88c/dev-pages/index.html#/light/app-layout/stateful?appLayoutToolbar=true then set the param to false and confirm, then set to classic and confirm. 

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__integ__/utils.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ import { viewports } from './constants';
 
 export const testIf = (condition: boolean) => (condition ? test : test.skip);
```
Comment (dpitcock) [inline src/app-layout/__integ__/utils.ts:10]: used to simplify debugging integ tests on local

**Review by jperals — DISMISSED** [review]

File: src/app-layout/visual-refresh-toolbar/drawer/local-drawer.tsx [hunk 2]
```diff
@@ -71,6 +71,7 @@ export function AppLayoutDrawerImplementation({ appLayoutInternals }: AppLayoutD
           aria-hidden={!activeDrawer}
           aria-label={computedAriaLabels.content}
           className={clsx(styles.drawer, sharedStyles['with-motion'], {
+            [styles['drawer-hidden']]: !activeDrawer,
```
Comment (georgylobko) [inline src/app-layout/visual-refresh-toolbar/drawer/local-drawer.tsx:74]: This css class is not needed here as this drawer in hidden state has `0px` width

File: src/app-layout/__tests__/runtime-drawers.test.tsx [hunk 3]
```diff
@@ -392,43 +392,39 @@ describeEachAppLayout(({ theme, size }) => {
       expect(onToolsChange).toHaveBeenCalledWith({ open: true });
     });
 
-    // Not implemented on the toolbar version yet
```
[the remark this answers is not in the document]
Comment (dpitcock) [reply src/app-layout/__tests__/runtime-drawers.test.tsx:395]: muscle memory.  resolved

File: src/app-layout/__integ__/awsui-applayout.test.ts [hunk 4]
```diff
@@ -121,6 +121,20 @@ describe.each(['classic', 'refresh', 'refresh-toolbar'] as Theme[])('%s', theme
     })
   );
 
+  test(
+    'preserves tools inner content state when drawer open and close',
```
Comment (georgylobko) [inline src/app-layout/__integ__/awsui-applayout.test.ts:125]: btw, do you see much value in this integ test? you've already covered this case in the unit test below, so they are basically duplicate each other

**Review by georgylobko — APPROVED** [review]

**Review by dpitcock — COMMENTED** [review]
I wonder if removing this line from the pr has anything to do with this https://github.com/cloudscape-design/components/blob/4057cd68ca0ddeae7858d3de629899209e58c49f/src/app-layout/visual-refresh-toolbar/drawer/local-drawer.tsx#L74

**Pull request merged**